app/services/search/buffer_suggestions_builder.rb

```diff
@@ -8,10 +8,20 @@ def initialize(location, search_params)
   end
 
   def get_buffers_suggestions
-    buffer_vacancy_count = LocationPolygon.with_name(location).buffers.map do |buffer|
-      [buffer.first, Search::AlgoliaSearchRequest.new(search_params.merge(polygon: [buffer.last])).stats.last]
-    end
+    if DOWNCASE_COMPOSITE_LOCATIONS.include?(location.downcase)
+      polygons = COMPOSITE_LOCATIONS[location.downcase].map do |component_region_name|
+        LocationPolygon.find_by(name: component_region_name.downcase, location_type: "counties")
+      end
 
+      buffer_vacancy_count = ImportPolygons::BUFFER_DISTANCES_IN_MILES.map do |distance|
+        buffers = polygons.map { |polygon| polygon.buffers[distance.to_s] }
+        [distance.to_s, Search::AlgoliaSearchRequest.new(search_params.merge(polygons: buffers)).stats.last]
+      end
+    else
+      buffer_vacancy_count = LocationPolygon.with_name(location).buffers.map do |buffer|
+        [buffer.first, Search::AlgoliaSearchRequest.new(search_params.merge(polygons: [buffer.last])).stats.last]
+      end
+    end
```

Comment:
My heart says we can trim this method down a bit, but my head hasn't caught up with me yet 😂

Reply:
hadnt quite scrolled to see connors comment but my first thought was also that the method is fairly long and could/should be broken down in to testable parts